dist/expected.go

```diff
@@ -0,0 +1,54 @@
+// Copyright 2016 The Go Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package dist
+
+import "github.com/gonum/quad"
+
+// ExpectedFixed numerically computes the expected value of the input function
+// under the probability distribution specified by the Quantiler. That
+// is, ExpectedFixed computes
+//
+//  ev ≈ E[f] = int_-inf^inf f(x) p(x) dx
+//
+// using the number of quadrature locations specified by evals. Concurrent
+// sets the number of allowed concurrent evaluations of f.
+//
+// ExpectedFixed is a wrapper around quad.Fixed. See the documentation for more
+// information.
+func ExpectedFixed(f func(float64) float64, q Quantiler, evals, concurrent int) (ev float64) {
```

Comment:
The discussion is definitely healthy. I do think you're right that fundamentally different algorithms deserve names, but I also think that if two things give the same answer in infinite precision they are the same algorithm. I don't think we can guarantee exact matching for floating point values always, and instead need to do what's reasonable. In the specific case, it's not so clear. What is the relative cost of 2*n float64 memory versus one mutex and locking?

I'm not sure we want that note. I agree that it's true, but it seems that it should be assumed in general that concurrency leads to non-determinstic ordering (unless stated explicitly otherwise), and that non-determinism in floating point addition leads to results that can differ slightly. In other words, the behavior you mention should be the default assumption for floating point numbers and concurrency.